lib/linux_admin/scap.rb

```diff
@@ -0,0 +1,67 @@
+require 'openscap'
+require 'nokogiri'
+
+module LinuxAdmin
+  class Scap
+    PROFILE_NAME = "temp-profile"
+    SSG_XML_PATH = "/usr/share/xml/scap/ssg/content/"
+
+    def lockdown(*args)
+      values = args.last.kind_of?(Hash) ? args.pop : {}
+      rules = args
+
+      detect_local_file_names
+
+      profile_xml = build_profile(PROFILE_NAME, "Temp Profile", "A Temp Profile", rules, values)
+      write_temp_xml(profile_xml)
+      run_openscap_fix
+      remove_temp_files
+    end
+
+    def build_profile(profile_id, title, desc, rules, values)
+      builder = Nokogiri::XML::Builder.new do |xml|
+        xml.Profile(:id => profile_id) do
+          xml.title(title)
+          xml.description(desc)
+          rules.each { |r| xml.select(:idref => r, :selected => "true") }
+          values.each { |k, v| xml.send("refine-value", :idref => k, :selector => v) }
+        end
+      end
+      builder.doc.root.to_xml
+    end
+
+    private
+
+    def local_ssg_files(type)
+      Dir["#{SSG_XML_PATH}ssg-*-#{type}.xml"].select { |f| f =~ /ssg-\w*-#{type}.xml/ }
+    end
+
+    def detect_local_file_names
+      @xccdf_file = File.basename(local_ssg_files("xccdf").first)
+      @oval_file = File.basename(local_ssg_files("oval").first)
+    end
+
+    def write_temp_xml(temp_profile_xml)
+      doc = Nokogiri::XML(File.open("#{SSG_XML_PATH}#{@xccdf_file}"))
+      model = doc.at_css("model")
+      model.add_next_sibling("\n#{temp_profile_xml}")
+      File.write("/tmp/#{@xccdf_file}", doc.root.to_xml)
+    end
+
+    def run_openscap_fix
+      FileUtils.cp("#{SSG_XML_PATH}#{@oval_file}", '/tmp')
+
+      session = OpenSCAP::Xccdf::Session.new("/tmp/#{@xccdf_file}")
+      session.load
+      session.profile = PROFILE_NAME
+      session.evaluate
+      session.remediate
+      session.destroy
```

Comment:
All of these lines could potentially raise an OpenSCAPError.  Is it better to let the caller of lockdown handle that or should we do something different?